Sources/XcodeGenKit/SourceGenerator.swift

```diff
@@ -286,9 +288,13 @@ class SourceGenerator {
             let isTopLevelGroup = (isBaseGroup && !createIntermediateGroups) || isRootPath
 
             let groupName = name ?? path.lastComponent
-            let groupPath = isTopLevelGroup ?
-                ((try? path.relativePath(from: project.basePath)) ?? path).string :
-                path.lastComponent
+            let groupPath: String
+            switch try? path.relativePath(from: projectDirectory ?? project.basePath).string {
+            case .some(let relativePath) where isTopLevelGroup:
+                groupPath = relativePath
+            default:
+                groupPath = path.lastComponent
```

Comment:
Could this be simplified to use an `if let`
```swift
if isTopLevelGroup, let relativePath = try? path.relativePath(from: projectDirectory ?? project.basePath).string {
    groupPath = relativePath
} else {
    groupPath = path.lastComponent
}
```

Reply:
There might be other places using `project.basePath` in the file. Perhaps we can have an instance property called `projectBasePath` that gets set in the initializer, which everything can read

Reply:
@yonaskolb I introduce `resolveGroupPath` instead. It may help us to get a good group path from projectBasePath relatively.

https://github.com/yonaskolb/XcodeGen/pull/681/commits/e63fe6be6f795109fd8de63911242c3b04211061